MinecraftClient/ChatBots/AutoAttack.cs

```diff
@@ -40,14 +55,53 @@ public override void Update()
                 attackCooldownCounter = attackCooldown;
                 if (entitiesToAttack.Count > 0)
                 {
-                    SendAnimation(Inventory.Hand.MainHand); // Arm animation
-                    foreach (KeyValuePair<int, Entity> entity in entitiesToAttack)
+                    if (!singleMode)
+                    {
+                        foreach (KeyValuePair<int, Entity> entity in entitiesToAttack)
+                        {
+                            // check that we are in range once again.
+                            bool shouldAttack = handleEntity(entity.Value);
+                            if (shouldAttack)
+                            {
+                                InteractEntity(entity.Key, 1); // hit the entity!
+                            }
+                        }
+                        SendAnimation(Inventory.Hand.MainHand); // Arm animation
+                    }
+                    else
                     {
-                        // check that we are in range once again.
-                        bool shouldAttack = handleEntity(entity.Value);
+                        int priorityEntity = 0;
+                        if (!priorityDistance) // low health priority
+                        {
+                            float health = int.MaxValue;
+                            foreach (var entity in entitiesToAttack)
+                            {
+                                if (entity.Value.Health < health)
+                                {
+                                    priorityEntity = entity.Key;
+                                    health = entity.Value.Health;
+                                }
+                            }
+                        }
+                        else // closest distance priority
+                        {
+                            double distance = 5;
+                            foreach (var entity in entitiesToAttack)
+                            {
+                                var tmp = GetCurrentLocation().Distance(entity.Value.Location);
+                                if (tmp < distance)
+                                {
+                                    priorityEntity = entity.Key;
+                                    distance = tmp;
+                                }
+                            }
+                        }
+                        // check entity distance and health again
+                        bool shouldAttack = handleEntity(entitiesToAttack[priorityEntity]);
```

Comment:
Here also you could convert into `if (shouldAttackEntity(entitiesToAttack[priorityEntity]))`